File: whsk2/lua/corelibs/Transform.lua
Transform = {}
Transform.__index = Transform

function Transform.new(position, rotation)

	position = position or Vector2.new()
	rotation = rotation or 0.0
	
	local transform = {}
	
	setmetatable(transform, Transform)
	
	transform.position = position
	transform.rotation = rotation
	
	return transform
	
end

return Transform

File: whsk2/Transform.cpp
#include "Transform.h"

using Engine::Transform;
using namespace Engine::Maths;

Transform::Transform(Vector2D pos, float rot)
	:
	position(pos),
	rotation(rot)
{

}

Transform::Transform()
	:
	Transform(Vector2D(), float())
{

}

Transform::~Transform()
{

}